javalib/src/main/scala/java/util/Properties.scala

```diff
@@ -51,11 +71,319 @@ class Properties(protected val defaults: Properties)
+                  put(key, value)
+                }
+                keyLength = -1
+                offset = 0
+                true
+              }
+            case '\\' =>
+              if (mode == KEY_DONE) keyLength = offset
+              mode = SLASH
+              isFirstChar
+            case ':' | '=' if keyLength == -1 =>
+              // if parsing the key
+              mode = NONE
+              keyLength = offset
+              isFirstChar
+            case _ =>
+              if (nextChar < 256 && Character.isWhitespace(nextChar)) {
+                if (mode == CONTINUE) mode = IGNORE
+                // if key length == 0 or value length == 0
+                if (offset == 0 || offset == keyLength || mode == IGNORE)
+                  isFirstChar
+                else if (keyLength == -1) {
+                  mode = KEY_DONE
+                  isFirstChar
+                } else {
+                  if (mode == IGNORE || mode == CONTINUE) mode = NONE
+                  else if (mode == KEY_DONE) {
+                    keyLength = offset
+                    mode = NONE
+                  }
+                  buf(offset) = nextChar
+                  offset += 1
+                  false
+                }
+              } else {
+                if (mode == IGNORE || mode == CONTINUE) mode = NONE
+                else if (mode == KEY_DONE) {
+                  keyLength = offset
+                  mode = NONE
+                }
+                buf(offset) = nextChar
+                offset += 1
+                false
+              }
+          }
+          if (mode == UNICODE) {
+            val digit = Character.digit(nextChar, 16)
+            if (digit >= 0) {
+              unicode = (unicode << 4) + digit
+              count += 1
+            } else if (count <= 4) {
+              throw new IllegalArgumentException(
+                "Invalid Unicode sequence: illegal character")
+            }
+            if (digit >= 0 && count < 4) {
+              isFirstChar
+            } else {
+              mode = NONE
+              buf(offset) = unicode.toChar
+              offset += 1
+              if (nextChar != '\n' && nextChar != '\u0085')
+                isFirstChar
+              else
+                fn(nextChar)
+            }
+          } else {
+            fn(nextChar)
+          }
+        }
+        processNext(_bool)
+      }
+    }
+
+    processNext(true)
+
+  }
+
+  private def writeComments(writer: Writer, comments: String): Unit = {
+    writer.write('#')
+    val chars = comments.toCharArray
+    var index = 0
+    while (index < chars.length) {
+      if (chars(index) < 256)
+        if (chars(index) == '\r' || chars(index) == '\n') {
+          val indexPlusOne = index + 1
+          if (chars(index) == '\r' && indexPlusOne < chars.length && chars(
+                indexPlusOne) == '\n') { // "\r\n"
+//          continue //todo: continue is not supported
```

Comment:
We don't leave TODOs in the code. They need to be addressed before this PR is merged.

Reply:
@densh will update this today